lib/reporters/TapReporter.js

```diff
@@ -116,7 +119,30 @@ function prettyYamlValue (value, indent = 4) {
   }
 
   // Handle null, boolean, array, and object
-  return JSON.stringify(value, null, 2);
+  return JSON.stringify(value, createCycleSafeReplacer(), 2);
+}
+
+/**
+ * Creates a replacer function to use with JSON.stringify that
+ * safely handles cyclical references. When a cycle is detected
+ * it will be represented as "[Circular]".
+ */
+function createCycleSafeReplacer() {
+  const seen = new Set();
```

Comment:
```suggestion
  const seen = new WeakSet();
```

Reply:
I made this change but then later didn't need it since I switched to using a shallow clone and the cloning is stack-based.